controller/officials.controller.go

```diff
@@ -175,3 +175,44 @@ func (h Handler) GetFormsByOfficialID(c *gin.Context) {
 		"data":     forms,
 	})
 }
+
+func (h Handler) VerifyUser(c *gin.Context) {
+	// Check the role of the user who sent the request
+	role := c.GetHeader("Role")
```

Comment:
instead of accessinf user via header and stuff accces the user via supertokens auth id then query the auth id in db find the user , check the user role . compare the user role to specific user role. 

this should be the flow